File: 563backpackV/backpackV.js
/**
 * @param nums: an integer array and all positive numbers
 * @param target: An integer
 * @return: An integer
 */
const backPackV = function (nums, target) {
    const dp = [1, ...Array(target).fill(0)]
    nums.forEach(item => {
        for (let size = target; size > item - 1; --size) dp[size] += dp[size - item]})
    return dp[dp.length - 1]
}

File: 563backpackV/backpackV.java
public class Solution {
    /**
     * @param nums: an integer array and all positive numbers
     * @param target: An integer
     * @return: An integer
     */
    public int backPackV(int[] nums, int target) {
        // write your code here
        final int[] dp = new int[target + 1];
        dp[0] = 1;
        Arrays.stream(nums).forEach(item -> java.util.stream.IntStream.range(item, target + 1).boxed().sorted(Comparator.reverseOrder()).forEach(size -> dp[size] += dp[size - item]));
        return dp[dp.length - 1];
    }
}
